yt_dlp/extractor/callin.py

```diff
@@ -0,0 +1,135 @@
+            'episode_number': 1,
+            'episode_id': '218b979630a35ead12c6fd096f2996c56c37e4d0dc1f6dc0feada32dcf7b31cd'
+        }
+    }]
+
+    def try_get_user_name(self, d):
+        names = [d.get(n) for n in ('first', 'last')]
+        if None in names:
+            return None
+        return ' '.join(names)
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        json_string = self._html_search_regex(
+            '<script\\s+id\\s*=\\s*["\']__NEXT_DATA__["\'].*?>(.+?)</script>',
+            webpage, 'json')
+        j = self._parse_json(json_string, display_id)
+        episode = j['props']['pageProps']['episode']
+
+        id = episode['id']
+        title = episode.get('title')
+        url = episode['m3u8']
+        formats = self._extract_m3u8_formats(url, id, ext='mp3')
+        self._sort_formats(formats)
+
+        show = traverse_obj(episode, ('show', 'title'))
+        show_id = traverse_obj(episode, ('show', 'id'))
+
+        # get the show metadata for some supplementary info
+        sj = None
+        try:
+            app_slug = self._html_search_regex(
+                '<script\\s+src=["\']/_next/static/([a-zA-Z0-9_]+)/_',
+                webpage, 'app slug', fatal=False)
+            if app_slug is None:
+                # this sometimes works, but sometimes seems to lag changes in the actual urls used.
+                app_slug = j['buildId']
+            show_slug = episode['show']['linkObj']['resourceUrl'].rsplit('/', 1)[1]
+            show_json_url = "https://www.callin.com/_next/data/" + app_slug + "/show/" + show_slug + ".json"
+
+            sj = self._download_json(show_json_url, id)
+        except (KeyError, IndexError, ExtractorError):
+            pass
+
+        host = traverse_obj(sj, ('pageProps', 'show', 'hosts', 0))
+        if host is None:
+            # guess that it's the first speaker listed
+            host = traverse_obj(episode, ('speakers', 0))
+
+        host_url = traverse_obj(sj, ('pageProps', 'show', 'url'))
+
+        host_nick = traverse_obj(host, ('linkObj', 'resourceUrl'))
+        host_nick = host_nick.rsplit('/', 1)[1] if (host_nick and '/' in host_nick) else None
+
+        categories = list(filter(None, [
+            c.get('name') for c in
+            traverse_obj(episode, ('show', 'categorizations')) or []
+        ]))
+
+        cast = list(filter(None, [
+            self.try_get_user_name(u) for u in
+            ((episode.get('speakers') or []) + (episode.get('callerTags') or []))
+        ]))
+
+        episode_number = None
+        if sj:
+            episode_list = traverse_obj(sj, ('pageProps', 'show', 'episodes')) or []
+            for (i, e) in enumerate(episode_list):
+                if e.get('id') == id:
+                    episode_number = len(episode_list) - i
+                    break
```

Comment:
```suggestion
        episode_list = traverse_obj(sj, ('pageProps', 'show', 'episodes')) or []
        episode_number = next(
            len(episode_list) - i for (i, e) in enumerate(episode_list)
            if e.get('id') == id, None)
```

Reply:
Done (in 5a94e184a2cd538ea64ad9b6c242056692be706c)